modules/ROOT/pages/client-credentials.adoc

```diff
@@ -0,0 +1,297 @@
+
+The annotation includes an expression that is applied on the response to extract the value. That value is then assigned to the field for the connection provider to use. When the connect(), validate() or disconnect() methods are invoked, the fields are set and usable.
+
+=== @ClientCredentialsState
+
+Every ConnectionProvider annotated with @ClientCredentials MUST contain one (and only one) field of type ClientCredentialsState.
+
+It is a simple immutable POJO that contains information regarding the outcome of the OAuth dance. It contains the following information:
+
+[source, java, linenums]
+----
+public interface ClientCrendentialsState {
+
+  /**
+   * @return The obtained access token
+   */
+  String getAccessToken();
+
+  /**
+   * @return The access token's expiration. The actual format of it depends on the OAuth provider
+   */
+  Optional<String> getExpiresIn();
+}
+----
+
+Through this object, the provider gains access to the accessToken and other standard information that was obtained during the OAuth dance. The original Salesforce example shows how the connect() method makes use of this POJO to create the client.
+
+== Configure a Client Credentials Connector
+
+An important aspect of OAuth-enabled modules is how to use them. As much as the SDK does to hide the complexities of the OAuth protocol, things like the OAuth dance are unavoidable. The SDK’s approach to this problem is to standardize the experience of all OAuth modules to keep simplifying the user experience.
+This section discusses the steps a user must take to use the module.
+
+=== Synthetic Parameters
+In addition to everything explicitly defined in the ConnectionProvider, the SDK automatically adds some extra parameters and injects the proper behavior for them.
+
+[cols="1,1,1,2,2", options="header"]
+|===
+|Parameter Name
+|Required
+|Expressions
+|Default Value
+|Description
+
+|`clientId`
+|YES
+|SUPPORTED
+|N/A
+|The OAuth `clientId` as registered with the service provider.
+
+|`clientSecret`
+|YES
+|NOT_SUPPORTED
+|N/A
+|The OAuth `clientSecret` as registered with the service provider.
+
+|`tokenUrl`
+|NO
+|SUPPORTED
+|Value provided in the @ClientCredentials annotation.
+|The service provider's token endpoint URL.
+
+|`scopes`
+|NO
+|SUPPORTED
+|Value provided in the @ClientCredentials annotation.
+|The OAuth scopes to be requested during the dance. If not provided, it defaults to those in the annotation.
+
+|`objectStore`
+|NO
+|NOT_SUPPORTED
+|N/A
+|A reference to the ObjectStore used for storing each resource owner ID’s data. If not specified, the runtime automatically provisions the default one.
+|===
+
+=== About the Use of Expressions
+
+The previous table identifies many of the synthetic parameters that accept expressions. Using expressions there has the same effect as using expressions in a regular parameter: it turns the configuration into a dynamic one.
```

Comment:
Avoid using locators like "previous," which create text section interdependencies and make potential reuse difficult.
Consider identifying by section heading:
```suggestion
The synthetic parameters table identifies many of those that accept expressions. Using expressions in synthetic parameters has the same effect as using expressions in regular parameters: they make the configuration dynamic.
```

Reply:
```suggestion
The synthetic parameters table identifies many of the synthetic parameters that accept expressions. Using expressions there has the same effect as using expressions in a regular parameter: it turns the configuration dynamic.
```